shopinvader_api_payment/routers/payment.py

```diff
@@ -0,0 +1,197 @@
+# Copyright Odoo SA (https://odoo.com)
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# @author Stéphane Bidoul <stephane.bidoul@acsone.eu>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+from typing import Annotated
+from urllib.parse import urljoin
+
+from fastapi import APIRouter, Depends, HTTPException, Request
+
+from odoo import api, models
+
+from odoo.addons.fastapi.dependencies import odoo_env
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.payment.models.payment_provider import PaymentProvider
+
+from ..schemas import (
+    PaymentDataWithProviders,
+    PaymentProvider as PaymentProviderSchema,
+    TransactionCreate,
+    TransactionProcessingValues,
+)
+
+_logger = logging.getLogger(__name__)
+
+payment_router = APIRouter(tags=["payment"])
+
+
+@payment_router.get("/payment/providers")
```

Comment:
```suggestion
@payment_router.get("/payment/methods")
```

... and all the implication on other things that needs to be renamed.

The reason is that we may want in the future to handle payment with tokens (i.e. card information recorded at the provider with a corresponding token stored in Odoo). The Odoo /payment/pay route does handle that.